include/extractor/traffic_flow_control_nodes.hpp

```diff
@@ -11,17 +11,29 @@ namespace osrm
 namespace extractor
 {
 
-struct TrafficSignals
+// The direction annotation is extracted from node tags.
+// The directions in which traffic flow object applies are relative to the way containing the node.
+enum class TrafficFlowControlNodeDirection : std::uint8_t
+{
+    NONE = 0,
+    ALL = 1,
+    FORWARD = 2,
+    REVERSE = 3
+};
+
+// represents traffic lights, stop signs, give way signs, etc.
+struct TrafficFlowControlNodes
```

Comment:
Not sure about naming here, if there are better ideas will be happy to discuss.

Reply:
Can we try to re-use the same mechanism in the future for example for railway crossings? https://wiki.openstreetmap.org/wiki/Tag:railway%3Dlevel_crossing